cmd/mtaresolver.go

```diff
@@ -0,0 +1,392 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"os"
+	"strings"
+
+	"github.com/SAP/cloud-mta/mta"
+	"github.com/joho/godotenv"
+)
+
+type MTAResolver struct {
+	mta.MTA
+	WorkingDir string
+	context    *ResolveContext
+}
+
+const RESOURCE_TYPE = 1
+const MODULE_TYPE = 2
+
+const VARIABLE_PREFIX = "~"
+const PLACEHOLDER_PREFIX = "$"
+const TEMPLATE_PREIX = "^"
+
+type MTASource struct {
+	Name       string
+	Parameters map[string]interface{} `yaml:"parameters,omitempty"`
+	Properties map[string]interface{} `yaml:"properties,omitempty"`
+	Type       int
+	Module     *mta.Module
+	Resource   *mta.Resource
+}
+
+type EnvVariables map[string]interface{}
+
+type Package struct {
+	Main    string                 `json:"main"`
+	Scripts map[string]interface{} `json:"scripts"`
+}
+
+func NewMTAResolver(m *mta.MTA, workspaceDir string) *MTAResolver {
+	resolver := &MTAResolver{*m, workspaceDir, &ResolveContext{
+		global:    map[string]string{},
+		modules:   map[string]map[string]string{},
+		resources: map[string]map[string]string{},
+	}}
+
+	for _, module := range m.Modules {
+		resolver.context.modules[module.Name] = map[string]string{}
+	}
+
+	for _, resource := range m.Resources {
+		resolver.context.resources[resource.Name] = map[string]string{}
+	}
+	return resolver
+}
+
+func (s *MTASource) IsResource() bool {
+	return s.Type == RESOURCE_TYPE
+}
+
+func (m *MTAResolver) ResolveProperies(module *mta.Module) {
+
+	if m.Parameters == nil {
+		m.Parameters = map[string]interface{}{}
+	}
+
+	//add env variables
+	for _, val := range os.Environ() {
+		pos := strings.Index(val, "=")
```

Comment:
I should suggest to use strings.SplitN function

Reply:
I prefer this way - more clear, as I always want the first